services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -847,7 +848,8 @@ def _column_from_feature(
             column_stats = column.compute_and_prepare_response(local_parquet_split_directory)
         else:
             try:
-                data = pl.read_parquet(local_parquet_split_glob, columns=[column.name])
+                table = pq.read_table(local_parquet_split_directory, columns=[column.name])
+                data: pl.DataFrame = pl.DataFrame._from_arrow(table)
```

Comment:
i should have used `pl.from_arrow()` instead of "accessing a protected member" in `pl.DataFrame._from_arrow()` but the former one returns `pl.DataFrame | pl.Series` instead of only `pl.DataFrame` which causes type check fail

Reply:
@severo actually seems that reading with arrow is 1.5 slower - absolute numbers are tenths of a second though. memory consumption is also higher idk why 
![image](https://github.com/huggingface/dataset-viewer/assets/16348744/957c2c5b-73c8-430a-9686-c407ca5aec59)

but i changed the code anyway to use native polars reader (as before) when possible [c190265](https://github.com/huggingface/dataset-viewer/pull/2686/commits/c1902656dc3e6759c728446386e2d23c6d21eb6d)

Reply:
Yes, they are often used outside of tests, too (e.g. you can find examples in `transformers`, `datasets`, `huggingface_hub`, etc.)

Reply:
it's also ok to do if not isinstance(...): raise TypeError to calm mypy instead of assert ...

Reply:
Yes, either this or explicitly tell `bandit` to ignore `assert` on this particular line:
```python
assert isinstance(data, pl.DataFrame)  # type narrowing  # nosec: B101
```